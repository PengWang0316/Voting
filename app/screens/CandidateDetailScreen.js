/**
 * @format
 * @flow
 */
import React from 'react';

import SafeAreaScreen from './SafeAreaScreen';
import CandidateDetail from '../components/CandidateDetail';

type Props = {
  navigation: Object,
};

export const CandidateDetailScreen = ({ navigation }: Props) => (
  <SafeAreaScreen>
    <CandidateDetail
      candidate={navigation.getParam('candidate')}
    />
  </SafeAreaScreen>
);

export default CandidateDetailScreen;
