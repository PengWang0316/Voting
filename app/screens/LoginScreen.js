import React, { Component } from 'react';
import SnackBar from '@kevinwang0316/react-native-snackbar-component';

import SafeAreaScreen from './SafeAreaScreen';
import LoginPanel from '../components/LoginPanel';
// import LoginScreenContext from '../contexts/LoginScreenContext';

type Props = {
  navigation: Object,
};

type States = {
  isShowSnackbar: boolean,
  snackbarMessage: string,
  snackbarBgColor: string,
  isShowRegisterPanel: boolean,
};

const AUTO_HIDDING_TIME = 1500;

/**
 * The component that contains login form and register form.
 */
export class LoginScreen extends Component<Props, States> {
  state = {
    isShowSnackbar: false,
    snackbarMessage: '',
    snackbarBgColor: '',
  };

  /**
   * Updating the states that relates to the snackbar.
   * @param {object} snackbarStates contains isShowSnackbar, snackbarMessage, and snackbarBgColor.
   * @return {null} No return.
   */
  snackbarUpdateCallback = snackbarStates => this.setState(
    snackbarStates,
    () => setTimeout(() => this.setState({ isShowSnackbar: false }), AUTO_HIDDING_TIME + 100),
  );

  /**
   * Render function for the component.
   * @return {jsx} Return jsx.
   */
  render() {
    const {
      isShowSnackbar, snackbarMessage, snackbarBgColor,
    } = this.state;
    const { navigation } = this.props;
    return (
      <SafeAreaScreen>
        <LoginPanel handleSnackbarUpdate={this.snackbarUpdateCallback} navigation={navigation} />
        <SnackBar
          visible={isShowSnackbar}
          textMessage={snackbarMessage}
          backgroundColor={snackbarBgColor}
          autoHidingTime={AUTO_HIDDING_TIME}
        />
      </SafeAreaScreen>
    );
  }
}

export default LoginScreen;
