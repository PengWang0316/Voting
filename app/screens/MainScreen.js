import React, { Component } from 'react';
import { View, StyleSheet } from 'react-native';
import { connect } from 'react-redux';

import SafeAreaScreen from './SafeAreaScreen';
import CandidateWithBackground from '../components/CandidateWithBackground';
import { fetchCandidatesInfo } from '../actions/CandidateActions';

type Props = {
  candidates: Object,
  fetchCandidatesInfo: Function,
};

const CANDIDATE_DETAIL_PAGE = 'CandidateDetail';

const styles = StyleSheet.create({
  rootView: {
    width: '100%',
    height: '100%',
  },
  // menuCategoryContainer: {
  //   marginLeft: 'auto',
  //   marginRight: 'auto',
  //   marginTop: 20,
  //   width: '90%',
  //   paddingBottom: 200,
  // },
});

/**
 * Menu Screen component
 */
export class MainScreen extends Component<Props> {
  /**
   * Call the fetchCandidatesInfo action if Redux does not have candidates info yet.
   * @param {object} props has all component's prop value.
   */
  constructor(props) {
    super(props);
    if (!props.candidates) props.fetchCandidatesInfo();
  }

  /**
   * Navigate to the candidate detial page
   * @param {Object} candidate is an object that has all candidate's information
   * @return {null} No return.
   */
  handleClickCandidate = candidate => {
    this.props.navigation.navigate(CANDIDATE_DETAIL_PAGE, { candidate });
  };

  /**
   * The render method
   * @return {jsx} Return the jsx for the component.
   */
  render() {
    const { candidates } = this.props;
    return (
      <SafeAreaScreen>
        <View style={styles.rootView}>
          {candidates && candidates.map(candidate => (
            <CandidateWithBackground
              key={candidate.id}
              candidate={candidate}
              onClick={this.handleClickCandidate}
            />
          ))}
        </View>
      </SafeAreaScreen>
    );
  }
}
/* istanbul ingore next */
const mapStateToProps = state => ({ candidates: state.candidates });
/* istanbul ingore next */
const mapDispatchToProps = dispatch => ({
  fetchCandidatesInfo: () => dispatch(fetchCandidatesInfo()),
});
export default connect(mapStateToProps, mapDispatchToProps)(MainScreen);
