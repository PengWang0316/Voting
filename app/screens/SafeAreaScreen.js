import React from 'react';
import { SafeAreaView, StyleSheet } from 'react-native';

import DarkerStatusBar from '../components/DarkerStatusBar';
import Theme from '../Theme';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: Theme.panelBackgroundColor,
    width: '100%',
  },
});

type Props = {
  children: Object,
  style: Object,
};
/**
 * Wrap the children components to a safe area component with a customizedstatus bar.
 * @param {object} children has all component's children component.
 * @return {jsx} Return jsx for the component.
 */
export const SafeAreaScreen = ({ children, style }: Props) => (
  <SafeAreaView style={[styles.container, style]}>
    <DarkerStatusBar />
    {children}
  </SafeAreaView>
);
export default SafeAreaScreen;
