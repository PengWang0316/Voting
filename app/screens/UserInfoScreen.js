import React from 'react';
import { connect } from 'react-redux';
import { View, Text, StyleSheet } from 'react-native';
import { Avatar, Button } from 'react-native-elements';

import Theme from '../Theme';
import SafeAreaScreen from './SafeAreaScreen';
import { logout as logoutAction } from '../actions/UserActions';

type Props = {
  user: Object,
  navigation: Object,
  logout: Function,
};

const Styles = StyleSheet.create({
  view: {
    marginBottom: 35,
  },
  btnView: {
    marginBottom: 10,
  },
  usernameText: {
    fontSize: 20,
    fontWeight: 'bold',
    color: Theme.primary.main,
  },
  logoutBtn: {
    width: '100%',
    backgroundColor: Theme.dangerousColor,
  },
  closeBtn: {
    width: '100%',
    backgroundColor: Theme.primary.light,
  },
});

let propLogout;
let goBack;

/**
 * Calling the logout action to logout and goBack function to go back to the previous screen.
 * @return {null} No return.
 */
const handleLogoutBtnClick = () => {
  propLogout();
  goBack();
};

/**
 * Calling the goBack function to go back to the previous screen.
 * @return {null} No return.
 */
const handleCloseBtnClick = () => goBack();

/**
 * The component to show users picture and offerring a way to log out.
 * @param {Object} props contains props from its parent component and Redux action.
 * @return {jsx} Return jsx for the component.
 */
export const UserInfoScreen = ({ user, navigation, logout }: Props) => {
  propLogout = logout;
  goBack = navigation.goBack;
  return (
    <SafeAreaScreen>
      <View style={Styles.view}>
        <Avatar icon={{ type: 'font-awesome', name: 'user-secret' }} size={130} rounded />
      </View>
      <View style={Styles.view}>
        <Text style={Styles.usernameText}>{user.displayName}</Text>
      </View>
      <View style={Styles.btnView}>
        <Button buttonStyle={Styles.logoutBtn} title="LOGOUT" onPress={handleLogoutBtnClick} />
      </View>
      <View>
        <Button buttonStyle={Styles.closeBtn} title="CLOSE" onPress={handleCloseBtnClick} />
      </View>
    </SafeAreaScreen>
  );
};
/* istanbul ignore next */
const mapStateToProps = state => ({
  user: state.user,
});
/* istanbul ignore next */
const mapDispatchToProps = dispatch => ({
  logout: () => dispatch(logoutAction()),
});
export default connect(mapStateToProps, mapDispatchToProps)(UserInfoScreen);
