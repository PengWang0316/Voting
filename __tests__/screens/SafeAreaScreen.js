import React from 'react';
import renderer from 'react-test-renderer';
import { Text } from 'react-native';

import { SafeAreaScreen } from '../../app/screens/SafeAreaScreen';

jest.mock('SafeAreaView', () => 'SafeAreaView');
jest.mock('../../app/components/DarkerStatusBar', () => 'DarkerStatusBar');

describe('SafeAreaScreen Test', () => {
  test('Snapshot test', () => expect(renderer.create(<SafeAreaScreen><Text>text</Text></SafeAreaScreen>).toJSON()).toMatchSnapshot());
});
