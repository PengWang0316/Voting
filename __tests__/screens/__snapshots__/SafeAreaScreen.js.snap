// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SafeAreaScreen Test Snapshot test 1`] = `
<SafeAreaView
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#F5FCFF",
        "flex": 1,
        "justifyContent": "center",
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <DarkerStatusBar />
  <Text>
    text
  </Text>
</SafeAreaView>
`;
